pkg/controller/manager/sync.go

```diff
@@ -24,59 +25,78 @@ func (c *Controller) getAuthToken(sc *scyllav1.ScyllaCluster) (string, error) {
 	return helpers.GetAgentAuthTokenFromSecret(secret)
 }
 
-func (c *Controller) getManagerState(ctx context.Context, clusterID string) (*state, error) {
-	clusters, err := c.managerClient.ListClusters(ctx)
+func (c *Controller) getManagerClusterState(ctx context.Context, sc *scyllav1.ScyllaCluster) (*managerClusterState, error) {
+	managerClusters, err := c.managerClient.ListClusters(ctx)
 	if err != nil {
-		return nil, err
-	}
-	var (
-		repairTasks map[string]scyllav1.RepairTaskStatus
-		backupTasks map[string]scyllav1.BackupTaskStatus
-	)
-
-	if clusterID != "" {
-		clusterFound := false
-		for _, c := range clusters {
-			if c.ID == clusterID {
-				clusterFound = true
-			}
+		return nil, fmt.Errorf("can't list clusters registered with manager: %w", err)
+	}
+
+	var managerCluster *managerclient.Cluster
+	clusterName := naming.ManagerClusterName(sc)
+	for _, mc := range managerClusters {
+		// Cluster names in manager state are unique, so it suffices to only find one with a matching name.
+		if mc.Name == clusterName {
+			managerCluster = mc
+			break
 		}
+	}
+
+	if managerCluster == nil {
+		return &managerClusterState{}, nil
+	}
```

Comment:
could be replace with slices.Find which would be much shorter

Reply:
good point, fixed